### PR TITLE
8263659: Reflow GTestResultParser for better readability

### PR DESCRIPTION
**Pull request by alblue**
As noted by https://sonarcloud.io/code?id=shipilev_jdk&selected=shipilev_jdk%3Atest%2Fhotspot%2Fjtreg%2Fgtest%2FGTestResultParser.java there are a few fixes that can be applied for the GTestResultParser:

* Avoiding nested 'try' statements
* Avoiding nested 'switch' statements
* Adding a break for each switch case to prevent accidental/unwanted fall-through
* Disabling ability to load from remote files when parsing XML files

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263659](https://bugs.openjdk.java.net/browse/JDK-8263659): Reflow GTestResultParser for better readability

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/2991/head:pull/2991`
`$ git checkout pull/2991`

To update a local copy of the PR:
`$ git checkout pull/2991`
`$ git pull https://git.openjdk.java.net/jdk pull/2991/head`

### CODE REVIEW

File: test/hotspot/jtreg/gtest/GTestResultParser.java
Comment (shipilev) [inline test/hotspot/jtreg/gtest/GTestResultParser.java:51]: Let's write this as:

```
while (xmlReader.hasNext()) {
    int code = xmlReader.next();
    if (code == XMLStreamConstants.START_ELEMENT) {
        switch (xmlReader.getLocalName()) {
```

**Review by shipilev — CHANGES_REQUESTED** [review]
Please change this PR synopsis to "8263659: Reflow GTestResultParser for better readability" to get this hooked properly. Also, enable testing, see "Pre-submit test status" in "Checks".

**Comment by alblue** [comment]
One of the test jobs hasn't finished running according to the above, but has if you drill down into the check itself. Is that blocking this PR moving forwards? Should I try to re-run the checks?

**Comment by shipilev** [comment]
I think all checks have passed: https://github.com/openjdk/jdk/pull/2991/checks.
We just need a test maintainer to ack the patch. @iignatev, want to to take a look?

**Review by iignatev — APPROVED** [review]
LGTM. @alblue, thanks for fixing that.

**Review by shipilev — APPROVED** [review]

**Pull request closed without merging**
